### PR TITLE
API requests - batch 1

### PR DESCRIPTION
**Pull request by brentwatson**
Going to do multiple PRs for implementing api requests to avoid a massive PR.  This is batch No. 1.

### CODE REVIEW

File: api/src/main/java/com/percolate/sdk/api/request/activity/ActivityParams.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.Map;
+
+/**
+ * Parameters for Activity request.
+ */
+@SuppressWarnings("unused")
+public class ActivityParams {
+    private Map<String, Object> params = new HashMap<>();
+
+    public ActivityParams() {
```
Comment (msya) [inline api/src/main/java/com/percolate/sdk/api/request/activity/ActivityParams.java]: one one line?

File: api/src/main/java/com/percolate/sdk/api/PercolateApi.java [hunk 2]
```diff
@@ -78,7 +87,80 @@ public PercolateServer getSelectedServer() {
...
+    @SuppressWarnings("unused")
     public TermsRequest terms() {
         return new TermsRequest(this);
     }
+
+    /**
+     * @return {@link ApprovalPoolsRequest} instance.
+     */
+    @SuppressWarnings("unused")
+    public ApprovalPoolsRequest approvalPools() {
```
Comment (msya) [inline api/src/main/java/com/percolate/sdk/api/PercolateApi.java]: If more request are added here, would this file get long?

File: api/src/main/java/com/percolate/sdk/api/request/approvals/workflow/ApprovalWorkflowService.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import com.percolate.sdk.api.PercolateApi;
+import com.percolate.sdk.dto.Workflow;
+import retrofit2.Call;
+import retrofit2.http.GET;
+import retrofit2.http.QueryMap;
+
+import java.util.Map;
+
+/**
+ * Percolate v5/terms API definition.
```
Comment (msya) [inline api/src/main/java/com/percolate/sdk/api/request/approvals/workflow/ApprovalWorkflowService.java]: v3/approval/workflow

File: api/src/main/java/com/percolate/sdk/api/request/approvals/workflow/history/ApprovalWorkflowHistoryParams.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package com.percolate.sdk.api.request.approvals.workflow.history;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Parameters for Approval Workflow request.
```
Comment (msya) [inline api/src/main/java/com/percolate/sdk/api/request/approvals/workflow/history/ApprovalWorkflowHistoryParams.java]: Approval Workflow History

File: api/src/main/java/com/percolate/sdk/api/request/license/LicenseService.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import com.percolate.sdk.api.PercolateApi;
+import com.percolate.sdk.dto.Licenses;
+import retrofit2.Call;
+import retrofit2.http.GET;
+import retrofit2.http.QueryMap;
+
+import java.util.Map;
+
+/**
+ * Percolate v45/license API definition.
```
Comment (msya) [inline api/src/main/java/com/percolate/sdk/api/request/license/LicenseService.java]: v3/licenses

File: api/src/main/java/com/percolate/sdk/api/PercolateApi.java (hunk 2 above)
Comment (brentwatson) [inline api/src/main/java/com/percolate/sdk/api/PercolateApi.java]: Yes, and I think that's OK.  It allows a client to use auto-complete in their IDE to find all available endpoints.  This object is a factory-like class.

File: api/src/main/java/com/percolate/sdk/api/request/approvals/workflow/ApprovalWorkflowService.java (hunk 3 above)
Comment (brentwatson) [inline api/src/main/java/com/percolate/sdk/api/request/approvals/workflow/ApprovalWorkflowService.java]: :+1:

File: api/src/main/java/com/percolate/sdk/api/request/approvals/workflow/history/ApprovalWorkflowHistoryParams.java (hunk 4 above)
Comment (brentwatson) [inline api/src/main/java/com/percolate/sdk/api/request/approvals/workflow/history/ApprovalWorkflowHistoryParams.java]: :+1:

File: api/src/main/java/com/percolate/sdk/api/request/license/LicenseService.java (hunk 5 above)
Comment (brentwatson) [inline api/src/main/java/com/percolate/sdk/api/request/license/LicenseService.java]: :+1:

**Pull request merged**